inlang/source-code/cross-sell/cross-sell-sherlock/src/recommendation/utils/exists.test.ts

```diff
@@ -45,8 +45,10 @@ const mockFiles = (files: memfs.NestedDirectoryJSON) => {
 			// @ts-ignore - memfs has the same interface as node:fs/promises
 			vi.spyOn(_memfs.promises, prop).mockImplementation(lixFs[prop])
 		} else {
-			// @ts-ignore - memfs has the same interface as node:fs/promises
-			vi.spyOn(_memfs.promises, prop)
+			if (prop in _memfs.promises) {
```

Comment:
was this change intentional?

Reply:
Yes.

On newer versions of `node` there is a `"glob"` property on `fs/promises` which is not present on memory FS. This caused the test setup to fail. I modified it to only spy on the subset of `fs` that is supported by `memfs`